src/BuildScriptGenerator/Python/PythonPlatformDetector.cs

```diff
@@ -29,46 +29,52 @@ internal class PythonPlatformDetector : IPlatformDetector
         public PlatformDetectorResult Detect(RepositoryContext context)
         {
             var sourceRepo = context.SourceRepo;
-            if (!sourceRepo.FileExists(PythonConstants.RequirementsFileName)
-                && !sourceRepo.FileExists(PythonConstants.SetupDotPyFileName))
+
+            var isPythonApp = IsPythonApp(sourceRepo);
+
+            // This detects if a runtime.txt file exists and if that is a python file
+            var versionFromRuntimeFile = DetectPythonVersionFromRuntimeFile(context.SourceRepo);
+            if (!isPythonApp && string.IsNullOrEmpty(versionFromRuntimeFile))
             {
-                _logger.LogDebug($"'{PythonConstants.SetupDotPyFileName}' or '{PythonConstants.RequirementsFileName}' " +
-                    $"does not exist in source repo");
                 return null;
             }
-            else if (!sourceRepo.FileExists(PythonConstants.RequirementsFileName)
-                && sourceRepo.FileExists(PythonConstants.SetupDotPyFileName))
+
+            return new PlatformDetectorResult
             {
-                _logger.LogInformation($"'{PythonConstants.RequirementsFileName} doesn't exist in source repo.' " +
-                    $"Oryx will try to build from '{PythonConstants.SetupDotPyFileName}'that exists in source repo");
+                Platform = PythonConstants.PlatformName,
+                PlatformVersion = versionFromRuntimeFile,
+            };
+        }
+
+        private bool IsPythonApp(ISourceRepo sourceRepo)
+        {
+            if (sourceRepo.FileExists(PythonConstants.RequirementsFileName))
+            {
+                _logger.LogInformation($"Found {PythonConstants.RequirementsFileName} at the root of the repo.");
+                return true;
             }
             else
             {
-                _logger.LogInformation($"'{PythonConstants.SetupDotPyFileName} doesn't exist in source repo.' " +
-                    $"Oryx will try to build from '{PythonConstants.RequirementsFileName}'that exists in source repo");
+                _logger.LogInformation(
+                    $"Cound not find {PythonConstants.RequirementsFileName} at the root of the repo.");
             }
 
-            // This detects if a runtime.txt file exists if that is a python file
-            var versionFromRuntimeFile = DetectPythonVersionFromRuntimeFile(context.SourceRepo);
-            if (string.IsNullOrEmpty(versionFromRuntimeFile))
+            var files = sourceRepo.EnumerateFiles(PythonConstants.PythonFileNamePattern, searchSubDirectories: true);
```

Comment:
so there is no setup.py in the root ... and no req.txt in the root .. but there is app.py or something .. which successfully detects it as python app ... but I think build will fail .. can we have a test on this? even a fail test just to confirm the scenario

Reply:
Good point. Will add a test.